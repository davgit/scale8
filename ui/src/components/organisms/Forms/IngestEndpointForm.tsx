import { FC } from 'react';
import ControlledTextInput from '../../atoms/ControlledInputs/ControlledTextInput';
import DrawerFormLayout from '../../molecules/DrawerFormLayout';
import { IngestEndpointFormProps } from '../../../dialogPages/dataManager/IngestEndpointCreate';
import StorageProviderSelector from '../../molecules/StorageProviderSelector';
import CheckBoxInput from '../../atoms/InputTypes/CheckBoxInput';
import { Mode } from '../../../gql/generated/globalTypes';
import { useConfigState } from '../../../context/AppContext';

const IngestEndpointForm: FC<IngestEndpointFormProps> = (props: IngestEndpointFormProps) => {
    const { mode } = useConfigState();

    return (
        <DrawerFormLayout {...props}>
            <ControlledTextInput
                name="name"
                label="Name"
                formProps={props}
                className="DrawerFormField"
                required
                autoFocus
            />
            {mode !== Mode.COMMERCIAL && (
                <>
                    <CheckBoxInput
                        name="analyticsEnabled"
                        value={props.values.analyticsEnabled}
                        setValue={(v) => {
                            props.handleChange('analyticsEnabled', v);
                        }}
                        label="Enable Analytics"
                        className="DrawerFormField"
                        style={{ marginLeft: '-11px' }}
                        color="primary"
                    />
                    <small className="DrawerFormField" style={{ marginTop: '-24px' }}>
                        Enable analytics dashboard when using Data Manager. We recommend this option
                        is left enabled.
                    </small>
                    {props.values.analyticsEnabled && (
                        <StorageProviderSelector
                            {...props}
                            warnGraphDisabled
                            infoText="Your analytics will be sent to your chosen storage provider. MongoDB is a
                        great solution for very low traffic websites and doesn't require any
                        configuration unless you wish to specify your own MongoDB servers. We
                        strongly recommend using BigQuery or Clickhouse for larger websites."
                        />
                    )}
                </>
            )}
        </DrawerFormLayout>
    );
};

export default IngestEndpointForm;
