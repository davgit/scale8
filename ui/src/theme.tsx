import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
