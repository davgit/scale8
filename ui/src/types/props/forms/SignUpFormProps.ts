import { FormProps } from '../../../hooks/form/useFormValidation';
import { ApolloError } from '@apollo/client';
import { RefObject } from 'react';
import HCaptcha from '@hcaptcha/react-hcaptcha';

export type SignUpValues = {
    newPassword: string;
    newPasswordConfirm: string;
    domain: string;
    orgName: string;
    fullName: string;
    email: string;
    agree: boolean;
    CAPTCHAToken: string;
    tempAccessCode: string;
};

export type SignUpFormProps = FormProps<SignUpValues> & {
    gqlError?: ApolloError;
    success: boolean;
    email?: string;
    type: string;
    target: string | undefined;
    loading: boolean;
    qsEmail: string | undefined;
    captcha: RefObject<HCaptcha>;
};
