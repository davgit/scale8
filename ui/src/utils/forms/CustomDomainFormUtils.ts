import { ValidateConfiguration } from '../validators/validateFormValues';
import domainValidator from '../validators/domainValidator';
import { FormProps } from '../../hooks/form/useFormValidation';

export type CustomDomainValues = {
    domain: string;
    certificate: string;
    key: string;
};

export const CustomDomainValidators: ValidateConfiguration<CustomDomainValues>[] = [
    {
        field: 'domain',
        validator: domainValidator,
        error: () => 'Invalid domain',
        ignoreEmpty: true,
    },
];

export type CustomDomainFormProps = FormProps<CustomDomainValues> & {
    cname: string;
};
