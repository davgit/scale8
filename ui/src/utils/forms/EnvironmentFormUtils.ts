import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import urlValidator from '../validators/urlValidator';

export type EnvVariable = { key: string; value: string };

export type EnvironmentValues = {
    name: string;
    url: string;
    revisionId: string;
    variables?: EnvVariable[];
    comments: string;
};

const EnvironmentCreateValidators: ValidateConfiguration<EnvironmentValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Environment name too short',
    },
    {
        field: 'url',
        validator: urlValidator,
        error: () => 'Invalid URL',
        ignoreEmpty: true,
    },
];

const EnvironmentUpdateValidators: ValidateConfiguration<EnvironmentValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Environment name too short',
    },
    {
        field: 'url',
        validator: urlValidator,
        error: () => 'Invalid URL',
        ignoreEmpty: true,
    },
];

export { EnvironmentCreateValidators, EnvironmentUpdateValidators };
