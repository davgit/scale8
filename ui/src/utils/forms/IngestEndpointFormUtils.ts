import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import { StorageProviderFields, storageProviderValidators } from '../StorageProviderUtils';
import { IngestSchemaWizard } from '../../gql/generated/globalTypes';

export type IngestEndpointValues = StorageProviderFields & {
    name: string;
    wizard: IngestSchemaWizard | '';
    analyticsEnabled: boolean;
};

const IngestEndpointValidators: ValidateConfiguration<IngestEndpointValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Ingest Endpoint name too short',
    },
    ...storageProviderValidators,
];

export { IngestEndpointValidators };
