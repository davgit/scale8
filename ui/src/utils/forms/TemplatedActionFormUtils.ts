import { PlatformDataMapInput } from '../../types/DataMapsTypes';
import { PlatformActionPermissionInput } from '../../types/ActionPermissionsTypes';
import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import descriptionValidator from '../validators/descriptionValidator';
import requiredStringValidator from '../validators/requiredStringValidator';
import parsedCodeValidator from '../validators/parsedCodeValidator';
import { PlatformActionPermissionRequest } from '../../gql/generated/globalTypes';

export type TemplatedActionsValues = {
    name: string;
    description: string;
    icon: string;
    code: string;
    execRaw: boolean;
    platformDataMaps: PlatformDataMapInput[];
    permissionRequests: PlatformActionPermissionInput[];
};

const TemplatedActionValidators: ValidateConfiguration<TemplatedActionsValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Action name too short',
    },
    {
        field: 'description',
        validator: descriptionValidator,
        error: () => 'Description too short',
    },
    {
        field: 'name',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'code',
        validator: parsedCodeValidator,
        error: (result) => `Invalid Code${result === 0 || `, Error: ${result}`}`,
    },
    {
        field: 'code',
        validator: async (
            value: string | boolean | PlatformActionPermissionInput[] | PlatformDataMapInput[],
        ): Promise<-1 | 0> => {
            return (value as string).includes('success()') ? -1 : 0;
        },
        error: () => 'Action code must invoke "success()"',
    },
    {
        field: 'icon',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'description',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'code',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'permissionRequests',
        validator: async (
            value: string | boolean | PlatformActionPermissionInput[] | PlatformDataMapInput[],
        ): Promise<-1 | 0> => {
            return (value as PlatformActionPermissionInput[]).find((_) => {
                if (
                    _.active &&
                    (_.permission === PlatformActionPermissionRequest.READ_PAGE_URL ||
                        _.permission === PlatformActionPermissionRequest.READ_REFERRER_URL)
                ) {
                    return _.urlParts === undefined || _.urlParts.length === 0;
                }
                return false;
            }) === undefined
                ? -1
                : 0;
        },
        error: () => 'Select at least one.',
    },
    {
        field: 'icon',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
];

export { TemplatedActionValidators };
