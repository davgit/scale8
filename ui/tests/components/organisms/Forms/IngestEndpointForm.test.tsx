import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import IngestEndpointForm from '../../../../src/components/organisms/Forms/IngestEndpointForm';
import { IngestEndpointValues } from '../../../../src/utils/forms/IngestEndpointFormUtils';

const MockForm: FC = () => {
    const formProps = useMockFormProps<IngestEndpointValues>({
        name: '',
        wizard: '',
        analyticsEnabled: true,
        storageProvider: 'AWS_KINESIS',
        region: '',
    });

    return <IngestEndpointForm {...formProps} isCreate={true} />;
};

describe('IngestEndpointForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
