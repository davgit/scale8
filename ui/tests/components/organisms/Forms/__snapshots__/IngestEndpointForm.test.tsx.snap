// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngestEndpointForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-s92abg"
  >
    <div
      class="MuiBox-root css-70qvj9"
    >
      <div
        class="MuiBox-root css-1isemmb"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-uix0gv"
      >
        mock form
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-yn9cjk"
    >
      <div
        class="MuiBox-root css-1676va5"
      >
        <form
          class="MuiBox-root css-mjj15b"
        >
          <div
            class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused Mui-required css-1oyovms-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Name
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-adornedEnd css-1ejkwvn-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                name="name"
                required=""
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-18yztu3-MuiFormControlLabel-root"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-esq4az-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="useWizard"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
            >
              Use Wizard
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd DialogFormField css-fbaq5x-MuiFormControlLabel-root"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-esq4az-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="analyticsEnabled"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
            >
              Enable Analytics
            </span>
          </label>
          <small
            class="DialogFormField MuiBox-root css-mnf576"
          >
            Enable analytics dashboard when using Data Manager.
          </small>
          <div
            class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1oyovms-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Storage Provider
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-1ejkwvn-MuiInputBase-root-MuiInput-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-storageProvider"
                class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                id="mui-component-select-storageProvider"
                role="button"
                tabindex="0"
              >
                Amazon Kinesis
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="storageProvider"
                required=""
                tabindex="-1"
                value="AWS_KINESIS"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard DialogFormField css-1sd9cqf-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-17ii7u1-MuiSvgIcon-root"
                data-testid="InfoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-acap47-MuiAlert-message"
            >
              Your analytics will be sent to your chosen storage provider. MongoDB is a great solution for very low traffic websites and doesn't require any configuration unless you wish to specify your own MongoDB servers. We strongly recommend using BigQuery or Clickhouse for larger websites.
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard DialogFormField css-18ix9of-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-17ii7u1-MuiSvgIcon-root"
                data-testid="ReportProblemOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-acap47-MuiAlert-message"
            >
              If you use Kinesis as provider the graphs in the dashboard will be disabled.
            </div>
          </div>
          <div
            class="MuiBox-root css-17ck9e8"
          >
            <div
              class="MuiBox-root css-79elbk"
            >
              <div
                class="MuiBox-root css-rxw2ex"
              >
                AWS Kinesis Storage Config
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                Stream Name
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1ejkwvn-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  name="streamName"
                  required=""
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                      data-testid="ClearIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                Access Key Id
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1ejkwvn-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  name="accessKeyId"
                  required=""
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                      data-testid="ClearIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                Secret Access Key
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1ejkwvn-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  name="secretAccessKey"
                  required=""
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                      data-testid="ClearIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                Region
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-1ejkwvn-MuiInputBase-root-MuiInput-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="mui-component-select-region"
                  class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                  id="mui-component-select-region"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="region"
                  required=""
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-wpygli"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-xcbv7o-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
